### PR TITLE
Hackathon Deployment Pipeline

### PR DESCRIPTION
**Pull request by akshay-acharekar**
Deployment pipeline for Apptware Hackathon website:
Step 1: Pulling code from main branch
Step 2: Building Node Application
Step 3:  Configure AWS credentials
Step 4: Checking Server State
Step 5: Get Runner IP
Step 6: Whitelist runner IP Address
Step 7: Copy files to server
Step 8: Starting Application
Step 9: Revoke runner IP Address
Step 10: Notify Google Chat
Step 11: Notify Google Chat for failure
Step 12: Notify Google Chat for cancellation

### CODE REVIEW

File: .github/workflows/website-deployement.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        with:
+          ref: main
+
+      - name: Building node Application
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+        run: |
+          npm i
+          npm run buid
```
Comment (tejas-acharekar) [inline .github/workflows/website-deployement.yml:25]: Command: npm run build

File: .github/workflows/website-deployement.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      - name: Revoke runner ip address
+        if: ${{ steps.status.outputs.INSTANCE_STATE_UPDATE == 'running'}}
+        run: |
+          aws ec2 revoke-security-group-ingress \
+            --group-id ${{ secrets.AWS_INSTANCE_SG_ID }} \
+            --protocol tcp \
+            --port 22 \
+            --cidr ${{ steps.ip.outputs.ipv4 }}/32
+
+      - name: Notify Google Chat
```
Comment (tejas-acharekar) [inline .github/workflows/website-deployement.yml:86]: Remove all the steps below from this line

**Review by tejas-acharekar — CHANGES_REQUESTED** [review]
Check and update @akshay-acharekar

File: .github/workflows/website-deployement.yml (hunk 1 above)
Comment (akshay-acharekar) [reply .github/workflows/website-deployement.yml]: Done

File: .github/workflows/website-deployement.yml (hunk 2 above)
Comment (akshay-acharekar) [reply .github/workflows/website-deployement.yml]: Done

**Review by tejas-acharekar — APPROVED** [review]
Updated as per required

**Pull request merged**